Rust/mechanics_fast/src/lib.rs

```diff
@@ -37,6 +38,22 @@ fn compute_gravitational_acceleration_many_on_one(masses: &[f64], positions: &[V
     acceleration
 }
 
+fn compute_gravitational_acceleration_many_on_many(masses: &[f64], positions: &[Vector3<f64>], accelerations: &mut [Vector3<f64>]) -> Vec<Vector3<f64>> {
+        let mut computed_accelerations = Vec::with_capacity(masses.len());
+
+    for i in 0..masses.len() {
+        if let Some(body_mass) = masses.get(i) {
+            let accel = compute_gravitational_acceleration_many_on_one(masses, positions, i);
+            computed_accelerations.push(accel);
+        } else {
+            computed_accelerations.push(Vector3::new(0.0, 0.0, 0.0));
+        }
```

Comment:
Could you please explain under what circumstances we would want to enter zero in the list instead of calling `compute_gravitational_acceleration_many_on_one`? I don't understand what the `if` statement is checking.